code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -26,17 +41,215 @@
+			return 0
+		var/mob/living/carbon/C = G.affecting
+		C.visible_message("<span class='danger'>[user] dunks [C]'s face into [src]'s lava!</span>", \
+						"<span class='userdanger'>[user] dunks your face into [src]'s lava!</span>")
+		if(!C.stat)
+			C.emote("scream")
+		user.changeNext_move(CLICK_CD_MELEE)
+		C.apply_damage(30, BURN, "head") //30 fire damage because it's FUCKING LAVA
+		C.status_flags |= DISFIGURED //Your face is unrecognizable because it's FUCKING LAVA
+		return 1
+
+/obj/structure/cult/cultforge/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		to_chat(user, "<span class='warning'>The heat radiating from [src] pushes you back.</span>")
+		return
+	if(!anchored)
+		to_chat(user, "<span class='cultitalic'>You need to anchor [src] to the floor with a tome first.</span>")
+		return
+	if(cooldowntime > world.time)
+		to_chat(user, "<span class='cultitalic'>The magic in [src] is weak, it will be ready to use again in [getETA()].</span>")
+		return
+	var/choice =  input(user,"You study the schematics etched on the force...","Forge") as null|anything in list("Shielded Hardsuit", "Flagellant's Robe","Cultist Hardsuit")
+	var/pickedtype
+	var/otheritem //ie:helmet..
+	switch(choice)
+		if("Shielded Hardsuit")
+			pickedtype = /obj/item/clothing/suit/hooded/cultrobes/cult_shield
+		if("Flagellant's Robe")
+			pickedtype = /obj/item/clothing/suit/hooded/cultrobes/berserker
+		if("Cultist Hardsuit")
+			pickedtype = /obj/item/clothing/suit/space/cult
+			otheritem = /obj/item/clothing/head/helmet/space/cult
+	if(pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
+		cooldowntime = world.time + 2400
+		var/obj/item/N = new pickedtype(get_turf(src))
+		if(otheritem)
+			new otheritem(get_turf(src))
+		to_chat(user, "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>")
+
+
+/obj/structure/cult/cultpylon
+	name = "pylon"
+	desc = "A floating crystal that slowly heals those faithful to a cult."
+	icon_state = "pylon"
+	light_range = 5
+	light_color = "#3e0000"
+	health = 50 //Very fragile
+	death_message = "<span class='warning'>The pylon's crystal vibrates and glows fiercely before violently shattering!</span>"
+	death_sound = 'sound/effects/pylon_shatter.ogg'
+	var/heal_delay = 25
+	var/last_heal = 0
+	var/corrupt_delay = 50
+	var/last_corrupt = 0
+
+/obj/structure/cult/cultpylon/New()
+	processing_objects |= src
+	..()
+
+/obj/structure/cult/cultpylon/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/structure/cult/cultpylon/process()
+	if(!anchored)
+		return
+	if(last_heal <= world.time)
+		last_heal = world.time + heal_delay
+		for(var/mob/living/L in range(5, src))
+			if(iscultist(L) || istype(L, /mob/living/simple_animal/shade) || istype(L, /mob/living/simple_animal/hostile/construct))
+				if(L.health != L.maxHealth)
+					new /obj/effect/overlay/temp/heal(get_turf(src), "#960000")//not working
```

Comment:
I have a working heal overlay (same as this one) here:
https://github.com/Markolie/Paradise/commit/e5c4b78f47e70514b6f170d1c9fd3c06c972bd4a#diff-1b9774d7406ce1c308f62ee255569c28R96
Called here:
https://github.com/Markolie/Paradise/commit/e5c4b78f47e70514b6f170d1c9fd3c06c972bd4a#diff-75b263daf1ea9f84203b4504c9057e95R367

Make sure the icon is set to 'icons/effects/effects.dmi' and that the heal icon_state is present there.